src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/Preferences.java

```diff
@@ -0,0 +1,95 @@
+package net.sacredlabyrinth.phaed.dynmap.simpleclans;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import net.sacredlabyrinth.phaed.simpleclans.Clan;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+
+public class Preferences {
+    private static final File prefFile = new File(DynmapSimpleClans.getInstance().getDataFolder(), "preferences.json");
+    private static JsonArray preferences;
+    private final Clan clan;
+
+    public Preferences(@NotNull Clan clan) {
+        this.clan = clan;
+        if (preferences == null) {
+            loadPreferences();
+        }
+    }
+
+    public static void loadPreferences() {
+        try {
+            if (!prefFile.exists()) {
+                prefFile.createNewFile();
+            }
+            preferences = new Gson().fromJson(new InputStreamReader(Files.newInputStream(prefFile.toPath())), JsonArray.class);
+        } catch (Exception ex) {
+            DynmapSimpleClans.getInstance().getLogger().severe("An error happened while reading the preferences file");
+            ex.printStackTrace();
+        }
+
+        if (preferences == null) {
+            preferences = new JsonArray();
+        }
+    }
+
+    private static void savePreferences() {
```

Comment:
Is it called in the main thread?

Reply:
I think so.

Reply:
Could be inside an async task.
Or use clan#getFlags?

```java
Flags flags = new Flags(clan.getFlags());
flags.put("defaultIcon", "clanhome");
clan.setFlags(flags.toJSONString());
```

Reply:
Let's see... if I'll delete DSC, would SC remove the flag as well?

Reply:
It wouldn't, but it's not a big problem I think.